components/movementsensor/adxl345/adxl345.go

```diff
@@ -189,10 +235,90 @@ func NewAdxl345(
 			}
 		}
 	})
-
+	sensor.interruptsFound = make(map[string]int)
+	sensor.readInterrupts(sensor.cancelContext)
+	err = sensor.configureInterruptRegisters(ctx)
+	if err != nil {
+		sensor.cancelFunc()
+		return nil, err
+	}
+	ticksChan := make(chan board.Tick)
+	if sensor.echoInterrupt1 != nil {
+		sensor.startInterruptPolling(sensor.echoInterrupt1, ticksChan)
+	}
+	if sensor.echoInterrupt2 != nil {
+		sensor.startInterruptPolling(sensor.echoInterrupt2, ticksChan)
+	}
 	return sensor, nil
 }
 
+func (adxl *adxl345) startInterruptPolling(interrupt board.DigitalInterrupt, ticksChan chan board.Tick) {
+	utils.PanicCapturingGo(func() {
+		interrupt.AddCallback(ticksChan)
+		defer interrupt.RemoveCallback(ticksChan)
+
+		for {
+			select {
+			case <-adxl.cancelContext.Done():
+				return
+			case tick := <-ticksChan:
+				if tick.High {
+					func() {
+						adxl.readInterrupts(adxl.cancelContext)
+					}()
+				}
+			}
+		}
+	})
+}
+
+func getInterruptConfigurations(cfg *AttrConfig) map[byte]byte {
+	var intEnabled byte
+	var intMap byte
+
+	if cfg.FreeFall != nil {
+		intEnabled += interruptBitPosition[FreeFall]
+		intMap += interruptBitPosition[FreeFall] & byte(cfg.FreeFall.AccelerometerPin<<2)
+	}
+	if cfg.SingleTap != nil {
+		intEnabled += interruptBitPosition[SingleTap]
+		intMap += interruptBitPosition[SingleTap] & byte(cfg.SingleTap.AccelerometerPin<<6)
+	}
+
+	return map[byte]byte{IntEnableAddr: intEnabled, IntMapAddr: intMap}
+}
+
+// This returns a map from register addresses to data which should be written to that register to configure the interrupt pin.
```

Comment:
:+1: Great comment, thanks! Could we get a similar one for `getInterruptConfigurations`?